searchlib/src/tests/tensor/hnsw_index/hnsw_index_test.cpp

```diff
@@ -144,11 +144,20 @@ class HnswIndexTest : public ::testing::Test {
         if (exp_hits.size() == k) {
             std::vector<uint32_t> expected_by_docid = exp_hits;
             std::sort(expected_by_docid.begin(), expected_by_docid.end());
-            auto got_by_docid = index->find_top_k(k, qv, k);
+            auto got_by_docid = index->find_top_k(k, qv, k, 100100.25);
             for (idx = 0; idx < k; ++idx) {
                 EXPECT_EQ(expected_by_docid[idx], got_by_docid[idx].docid);
             }
         }
+        if ((rv.size() > 1) && (rv[0].distance < rv[1].distance)) {
```

Comment:
Currently it is very hard to tell that distance thresholding is actually tested. Consider moving lines 152-160 to a separate function with a descriptive name. And are we sure that all relevant cases are tested when appending this to existing tests? Consider adding an explicit test function to HnswIndexTest if not.